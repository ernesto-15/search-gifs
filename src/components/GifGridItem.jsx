import React from 'react';
import './styles/GifGridItem.css';

const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
    </div>
  );
};

export default GifGridItem;
