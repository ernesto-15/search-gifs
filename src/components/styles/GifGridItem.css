.card {
  /* border: 3px solid red; */
  border-radius: 20px;
  overflow: hidden;
  max-width: 360px;
  margin: 0 auto;
  width: 100%;
  height: 100%;
  /* display: flex;
  justify-content: center;
  align-items: center; */
}

.card p {
  padding: 5px;
  text-align: center;
}

.card img {
  min-height: 100%;
  height: 200px;
  width: 100%;
  margin: 0;
}
