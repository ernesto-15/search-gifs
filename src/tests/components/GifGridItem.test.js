import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem.jsx';

describe('tests in GifGridItem component', () => {
  const title = 'test gif';
  const url =
    'https://media1.giphy.com/media/l41lI4bYmcsPJX9Go/giphy.gif?cid=5387cba3j9e063tiwnc295ppe1z8kq1ua2oigkzvhjtu8gym&rid=giphy.gif';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should show component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have src and alt attributes', () => {
    const image = wrapper.find('img');
    expect(image.prop('src')).toBe(url);
    expect(image.prop('alt')).toBe(title);
  });

  test('should have class animate__fadeIn', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('animate__fadeIn')).toBeTruthy();
  });
});
