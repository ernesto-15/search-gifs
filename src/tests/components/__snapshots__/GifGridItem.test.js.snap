// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in GifGridItem component should show component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test gif"
    src="https://media1.giphy.com/media/l41lI4bYmcsPJX9Go/giphy.gif?cid=5387cba3j9e063tiwnc295ppe1z8kq1ua2oigkzvhjtu8gym&rid=giphy.gif"
  />
</div>
`;
